### PR TITLE
Add PWM support for Renesas RA6, RA4, RA2 devices

### PR DESCRIPTION
**Pull request by khoa-nguyen-18**
Enable PWM support on these boards:
- RA6: ek_ra6e2, ek_ra6m1, ek_ra6m2, ek_ra6m3, ek_ra6m4, ek_ra6m5, fpb_ra6e1, fpb_ra6e2
- RA4: ek_ra4e2, ek_ra4m2, ek_ra4m3, ek_ra4w1
- RA2: ek_ra2a1

### CODE REVIEW

**Comment by khoa-nguyen-18** [comment]
Just rebase main to solve twister test fail

File: dts/bindings/pwm/renesas,ra-pwm.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Copyright (c) 2024 Renesas Electronics Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-description: Renesas RA8 Pulse Width Modulation
+description: Renesas RA Pulse Width Modulation
 
-compatible: "renesas,ra8-pwm"
+compatible: "renesas,ra-pwm"
```
Comment (soburi) [inline dts/bindings/pwm/renesas,ra-pwm.yaml:6]: This change makes it necessary to update the migration guide.

**Review by thaoluonguw — DISMISSED** [review]
It looks good to me.

**Review by bjarki-andreasen — DISMISSED** [review]

**Comment by bjarki-andreasen** [comment]
could you rebase and add an entry in the migration guide regarding the updated naming of ra8 to ra?

**Comment by khoa-nguyen-18** [comment]
Currently the test `pwm_loopback` has an issue. We're hopeful this PR will be merged soon: https://github.com/zephyrproject-rtos/zephyr/pull/82623

**Review by soburi — DISMISSED** [review]

**Comment by khoa-nguyen-18** [comment]
> could you rebase and add an entry in the migration guide regarding the updated naming of ra8 to ra?

We have added a commit to update the naming of Renesas PWM compatible from `renesas,ra8-pwm` to `renesas,ra-pwm`. Thanks you.

**Review by KhiemNguyenT — DISMISSED** [review]

**Review by bjarki-andreasen — DISMISSED** [review]
Nice!

**Review by soburi — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]

**Pull request merged**